### ISSUE TITLE
Changing namespace in kubeconfig creates confusion

### ISSUE THREAD

**Issue by dtimm** [k14s/kapp#79]
When developing a Kubernetes app deployed with `kapp`, a lot of developers tend to switch kubeconfig namespaces regularly to track pods and other resources through their lifecycle. This causes a lot of confusion with `kapp deploy`, namely a lot of the following:
```
Resource 'customresourcedefinition/policies.authentication.istio.io (apiextensions.k8s.io/v1beta1) cluster' is associated with a different label value 'kapp.k14s.io/app=1583442602912040000
```

There are many ways this could be clarified, but my preference would be to store and reference the deployment namespace of created resources. This would enable descriptive error messages along the lines of `This resource was deployed in the "default" namespace, you are currently targeting the "test" namespace"`.

**Comment by pianohacker** [comment]
I've run into this as well; I tend to target the `cf-system` namespace when I'm trying to debug issues within cf-for-k8s, and the errors that result from this are fairly opaque. If you're unaware of kapp storing state in your targeted namespace, it becomes impossible to delete the existing app deploy without manually finding the kapp label.

**Comment by cppforlife** [comment]
> it becomes impossible to delete the existing app deploy without manually finding the kapp label.

does `kapp ls -A` (or long form `kapp ls --all-namespaces`) help?

> There are many ways this could be clarified, but my preference would be to store and reference the deployment namespace of created resources. This would enable descriptive error messages along the lines

i dont think we need to store namespace on created resources (since that would be a lot of dup info); however, but i do agree error message could be improved. there is an issue that someone else filed about that as well: https://github.com/k14s/kapp/issues/33

**Comment by cppforlife** [comment]
added more info to error message:

```
kapp: Error: Ownership errors:
- Resource 'rolebinding/cf-workloads-app-rolebinding (rbac.authorization.k8s.io/v1) namespace: cf-workloads' is already associated with a different app 'cf' namespace: default (label 'kapp.k14s.io/app=1582578662724084000')
```

will be available in v0.22.0

**Comment by cppforlife** [comment]
v0.22.0 was shipped. closing, feel free to reopen if further improvement is desired.
